src/plots/cartesian/select.js

```diff
@@ -208,21 +208,23 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
 
     dragOptions.doneFn = function(dragged, numclicks) {
         corners.remove();
-        throttle.clear(throttleID);
-
-        if(!dragged && numclicks === 2) {
-            // clear selection on doubleclick
-            outlines.remove();
-            for(i = 0; i < searchTraces.length; i++) {
-                searchInfo = searchTraces[i];
-                searchInfo.selectPoints(searchInfo, false);
-            }
+        throttle.done(throttleID).then(function() {
+            throttle.clear(throttleID);
```

Comment:
Maybe `throttle.done()` should call `throttle.clear()` before returning a promise?

Reply:
> Maybe `throttle.done()` should call `throttle.clear()` before returning a promise?

I don't think so. It doesn't come into play here, but you could imagine wanting to finish whatever event had been throttled when `done` was called, without cutting off throttling of future events in the train. I guess that's saying I didn't mean `done` to mean you were necessarily done forever, just done with whatever is in the queue right now.

Reply:
Got it. Thanks!